disttask/framework/framework_err_handling_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework_test
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+
+	"github.com/pingcap/tidb/disttask/framework/dispatcher"
+	"github.com/pingcap/tidb/disttask/framework/proto"
+	"github.com/pingcap/tidb/disttask/framework/scheduler"
+	"github.com/pingcap/tidb/domain/infosync"
+	"github.com/pingcap/tidb/testkit"
+)
+
+type planErrFlowHandle struct {
+	callTime int
+}
+
+var _ dispatcher.TaskFlowHandle = (*planErrFlowHandle)(nil)
+
+func (*planErrFlowHandle) OnTicker(_ context.Context, _ *proto.Task) {
+}
+
+func (p *planErrFlowHandle) ProcessNormalFlow(_ context.Context, _ dispatcher.TaskHandle, gTask *proto.Task) (metas [][]byte, err error) {
```

Comment:
recommend mock this interface using mockgen, see `mock_lightning:` in makefile

Reply:
> FlowHandle's interface is unstable

the mock should be generated any time the interface it's updated

Reply:
> > FlowHandle's interface is unstable
> 
> the mock should be generated any time the interface it's updated

Oh I got it. I will try it in next pr. Since there is a big pr using the same way to mock flowHandle is in review, I don't want introduce too much conflict.